tor/onionfile/onion_file.go

```diff
@@ -0,0 +1,111 @@
+package onionfile
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/lnencrypt"
+	"github.com/lightningnetwork/lnd/tor"
+)
+
+var (
+	// ErrEncryptedTorPrivateKey is thrown when a tor private key is
+	// encrypted, but the user requested an unencrypted key.
+	ErrEncryptedTorPrivateKey = errors.New("it appears the Tor private key " +
+		"is encrypted but you didn't pass the --tor.encryptkey flag. " +
+		"Please restart lnd with the --tor.encryptkey flag or delete " +
+		"the Tor key file for regeneration")
+)
+
+// OnionFile is a file-based implementation of the OnionStore interface that
+// stores an onion service's private key.
+type OnionFile struct {
+	privateKeyPath string
+	privateKeyPerm os.FileMode
+	encryptKey     bool
+	keyRing        keychain.KeyRing
+}
+
+// A compile-time constraint to ensure OnionFile satisfies the OnionStore
+// interface.
+var _ tor.OnionStore = (*OnionFile)(nil)
+
+// NewOnionFile creates a file-based implementation of the OnionStore interface
+// to store an onion service's private key.
+func NewOnionFile(privateKeyPath string, privateKeyPerm os.FileMode,
```

Comment:
Now we still have two `OnionFile` structs. Can't this be integrated into the existing one? It seems we just add two new parameters and some additional logic. Or is the idea to remove the `tor.OnionFile` since it's not used anymore? I think the diff could be way smaller if we just integrated this new code into the existing struct.

Reply:
Yes deleted original onionfile in the Tor package. Moved the onionfile to a new package based on the convo in here: https://github.com/lightningnetwork/lnd/pull/4458#discussion_r620729177

Reply:
Ooooohhhh, I see now. Yeah I never deleted the OnionFile data in cmd_onion. Will fix that !